compiler/src/iree/compiler/Codegen/Common/MaterializeTuningSpecsPass.cpp

```diff
@@ -123,14 +123,28 @@ getDefaultTuningSpec(ModuleOp module,
     return failure();
   }
 
-  // Try to look up the default tuning spec for this architecture, if any.
-  StringRef arch = gpuTarget.getArch();
-  std::string defaultTuningSpecName =
-      llvm::formatv("iree_default_tuning_spec_{}.mlir", arch);
+  std::optional<StringRef> sku = gpuTarget.getSKU();
+  std::string defaultTuningSpecName;
   std::optional<StringRef> defaultTuningSpecSource;
-  EmbeddedDataDirectory::withGlobal([&](EmbeddedDataDirectory &dir) {
-    defaultTuningSpecSource = dir.getFile(defaultTuningSpecName);
-  });
+  if (sku) {
+    // Try to look up the default tuning spec for this sku.
+    defaultTuningSpecName =
+        llvm::formatv("iree_default_tuning_spec_{}.mlir", sku);
+    EmbeddedDataDirectory::withGlobal([&](EmbeddedDataDirectory &dir) {
+      defaultTuningSpecSource = dir.getFile(defaultTuningSpecName);
+    });
+  }
+  if (!defaultTuningSpecSource) {
+    // If SKU-specific spec is not found, fall back to the default
+    // architecture-based tuning spec.
+    StringRef arch = gpuTarget.getArch();
+    defaultTuningSpecName =
+        llvm::formatv("iree_default_tuning_spec_{}.mlir", arch);
+    EmbeddedDataDirectory::withGlobal([&](EmbeddedDataDirectory &dir) {
+      defaultTuningSpecSource = dir.getFile(defaultTuningSpecName);
+    });
+  }
```

Comment:
This function is getting a bit long, I think it'd be better to outline this code to a helper function.